DQM/SiStripMonitorClient/plugins/SiStripBadComponentInfo.cc

```diff
@@ -0,0 +1,231 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+
+#include "DQM/SiStripCommon/interface/SiStripFolderOrganizer.h"
+#include "DQM/SiStripMonitorClient/plugins/SiStripBadComponentInfo.h"
+#include "DQM/SiStripMonitorClient/interface/SiStripUtility.h"
+#include "CalibTracker/Records/interface/SiStripQualityRcd.h"
+
+#include <iostream>
+#include <iomanip>
+#include <cstdio>
+#include <string>
+#include <sstream>
+#include <cmath>
+
+//
+// -- Contructor
+//
+SiStripBadComponentInfo::SiStripBadComponentInfo(edm::ParameterSet const& pSet) : 
+    m_cacheID_(0),
+    bookedStatus_(false),
+    nSubSystem_(6),
+    qualityLabel_(pSet.getParameter<std::string>("StripQualityLabel"))
+{ 
+  // Create MessageSender
+  LogDebug( "SiStripBadComponentInfo") << "SiStripBadComponentInfo::Deleting SiStripBadComponentInfo ";
+}
+//
+// -- Destructor
+//
+SiStripBadComponentInfo::~SiStripBadComponentInfo() {
+  LogDebug("SiStripBadComponentInfo") << "SiStripBadComponentInfo::Deleting SiStripBadComponentInfo ";
+
+}
+//
+// -- Begin Run
+//
+void SiStripBadComponentInfo::beginRun(edm::Run const& run, edm::EventSetup const& eSetup) {
+  LogDebug ("SiStripBadComponentInfo") <<"SiStripBadComponentInfo:: Begining of Run";
+
+  //Retrieve tracker topology from geometry
+  eSetup.get<TrackerTopologyRcd>().get(tTopoHandle_);
+
+  unsigned long long cacheID = eSetup.get<SiStripQualityRcd>().cacheIdentifier();
+  if (m_cacheID_ == !cacheID) { 
+    
+    m_cacheID_ = cacheID; 
+    LogDebug("SiStripBadComponentInfo") <<"SiStripBadchannelInfoNew::readCondition : "
+				   << " Change in Cache";
+    eSetup.get<SiStripQualityRcd>().get(qualityLabel_,siStripQuality_);
+
+  }    
+}
+//
+// -- Read Condition
+//
+void SiStripBadComponentInfo::checkBadComponents() {
+  const TrackerTopology* const topo = tTopoHandle_.product();
```

Comment:
hi @fabiocos - i don't believe this conditions usage is valid at  end-of-job time.

Reply:
EventSetup components are not guaranteed to be available at endJob. That is why the `endJob` member function is not passed an `edm::EventSetup` object.